apps/hash-graph/lib/graph/src/store/postgres/ontology/entity_type.rs

```diff
@@ -25,197 +25,183 @@ impl<C: AsClient> PostgresStore<C> {
+                            &entity_type_vertex_id,
+                            OntologyEdgeKind::ConstrainsPropertiesOn,
+                        ),
+                        &temporal_axes,
+                    )
+                        .await?
+                    {
+                        let property_type_vertex_id = property_type.vertex_id(time_axis);
 
                         subgraph.insert_edge(
-                            &entity_type_id,
+                            &entity_type_vertex_id,
                             OntologyEdgeKind::ConstrainsPropertiesOn,
                             property_type_vertex_id.clone(),
                         );
 
-                        self.traverse_property_type(
-                            vec![property_type_vertex_id],
-                            temporal_axes.clone(),
+                        subgraph.insert_vertex(&property_type_vertex_id, property_type);
+
+                        property_type_queue.push((
+                            property_type_vertex_id,
                             GraphResolveDepths {
                                 constrains_properties_on: OutgoingEdgeResolveDepth {
-                                    outgoing: current_resolve_depths
-                                        .constrains_properties_on
-                                        .outgoing
+                                    outgoing: graph_resolve_depths.constrains_properties_on.outgoing
                                         - 1,
-                                    ..current_resolve_depths.constrains_properties_on
+                                    ..graph_resolve_depths.constrains_properties_on
                                 },
-                                ..current_resolve_depths
+                                ..graph_resolve_depths
                             },
-                            traversal_context,
-                            subgraph,
-                        )
-                        .await?;
+                            temporal_axes.clone()
+                        ));
                     }
                 }
 
-                if let Some(inherits_from_type_ref_urls) = inherits_from_type_ref_urls {
-                    for inherits_from_type_ref_url in inherits_from_type_ref_urls {
-                        let inherits_from_type_vertex_id =
-                            EntityTypeVertexId::from(inherits_from_type_ref_url);
+                if graph_resolve_depths.inherits_from.outgoing > 0 {
+                    for referenced_entity_type in <Self as Read<EntityTypeWithMetadata>>::read(
+                        self,
+                        &Filter::<EntityTypeWithMetadata>::for_ontology_edge_by_entity_type_vertex_id(
+                            &entity_type_vertex_id,
+                            OntologyEdgeKind::InheritsFrom,
+                            true,
```

Comment:
Why is this reversed?

Reply:
Let's say we have an entity type `person` and `friendship` with the edge `person --> friendship`. `person` has the link `friendship` (the destination does not matter here but the idea is the same). To get the `friendship` entity type we need to query for the entity type, where the reversed edge (`friendship --< person`)has the vertex id of the person: "give me all entity types, where the entity types, which uses _this_ entity type as a link have the vertex id of `person`".

Reply:
This is about `Inherits` from but I presume the logic is the same.

Are the knowledge graph filters written the other way around?
https://github.com/hashintel/hash/blob/a30bc9e4ea3b8b83dc0b5cdefa9d7c16256391da/apps/hash-graph/lib/graph/src/store/postgres/knowledge/entity.rs#L123-L130

Reads as if it were the other way around.

`outgoing_link_entity` is equivalent to a (kind=left_entity, reversed=true) relationship
- You are querying for `left_entity, true`
`inherits_from` here is equivalent to a (kind=inherits_from, reversed=false) relationship
- You are querying for `inherits_from, false`

Reply:
> This is about InheritsFrom not links but I presume the logic is the same.

Sorry, I already moved the comment to the other section but GitHub probably has not updated for you since then and you were already replying.

> Are the knowledge graph filters written the other way around?

This is a mistake I'm sure is not present in the follow-up. I have noticed this issue when writing the follow-up but as I just rebased this branch I have not yet updated this branch to use the correct logic. Thanks for pointing out, though. finding this is hard :D

Reply:
> Sorry, I already moved the comment to the other section

(Just for persistence) https://github.com/hashintel/hash/pull/2223#discussion_r1144646196

> This is a mistake I'm sure is not present in the follow-up. I have noticed this issue when writing the follow-up but as I just rebased this branch I have not yet updated this branch to use the correct logic. Thanks for pointing out, though. finding this is hard :D

Which is why I'm suggesting we might want to find a way to greatly improve the readability of these things https://github.com/hashintel/hash/pull/2223#discussion_r1144656382

Reply:
And this is exactly the reason why I want to abstract the logic away and don't have this logic in 11 places (each edge kind)